### PR TITLE
Feat/homepage revamp

### PR DESCRIPTION
**Pull request by zencephalon**
**What changed? Why?**

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/base-docs/tutorials/data.json [hunk 1]
```diff
@@ -1132,6 +1132,28 @@
...
+        "tags": [
+            "frontend"
+        ],
+        "difficulty": "intermediate",
+        "hide_table_of_contents": false,
+        "displayed_sidebar": null,
+        "last_updated": "Oct 7, 2024",
+        "duration": "10 min read",
+        "checksum": "a81681ca9176774c2b8d20b6a2a873bdbaab99944a4d90526797f699db3e7b86"
+    },
```
Comment (JFrankfurt) [inline apps/base-docs/tutorials/data.json:1156]: this looks like a mistaken inclusion

File: apps/web/global.d.ts [hunk 2]
```diff
@@ -27,4 +27,14 @@ declare module 'next/config' {
...
+
+declare module '*.glb' {
+  const src: string;
+  export default src;
+}
+
+declare module '*.gltf' {
+  const src: string;
+  export default src;
+}
```
Comment (JFrankfurt) [inline apps/web/global.d.ts:40]: wat is this

File: apps/web/package.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
     "analyze": "ANALYZE=true next build",
     "dev": "node --require ./tracer/initialize.js ./node_modules/.bin/next dev",
     "lint": "next lint",
-    "update-contributors": "node ./scripts/updateContributors.js && npx prettier ./src/components/CoreContributors/CoreContributors.json -w"
+    "update-contributors": "node ./scripts/updateContributors.js && npx prettier ./src/components/CoreContributors/CoreContributors.json -w",
+    "fetch-mirror-blog": "node ./scripts/pullLatestBlogPosts.js"
```
Comment (JFrankfurt) [inline apps/web/package.json:13]: what was our prior pattern? manual? why is this necessary for the new version?

File: apps/web/public/draco/draco_wasm_wrapper.js [hunk 4]
```diff
@@ -0,0 +1,2426 @@
+var $jscomp = $jscomp || {};
```
Comment (JFrankfurt) [inline apps/web/public/draco/draco_wasm_wrapper.js:1]: what makes this necessary? feels highly abnormal

File: apps/web/src/components/Button/ButtonWithLink.tsx [hunk 5]
```diff
@@ -22,10 +22,10 @@ export function ButtonWithLink({
   size,
   children,
   buttonClassNames,
-  rounded = false,
+  roundedFull = false,
   fullWidth = false,
   disabled = false,
-  isLoading = false,
+
```
Comment (JFrankfurt) [inline apps/web/src/components/Button/ButtonWithLink.tsx:28]: surprised to see this api change in this pr

File: apps/base-docs/tutorials/data.json (hunk 1 above)
Comment (kirkas) [reply apps/base-docs/tutorials/data.json]: fixed

File: apps/web/global.d.ts (hunk 2 above)
Comment (kirkas) [reply apps/web/global.d.ts:40]: tells typescript not to complain when you import custom files.

File: apps/web/package.json (hunk 3 above)
Comment (kirkas) [reply apps/web/package.json:13]: no prio pattern to load blog post, this is a helpful tasks to keep homepage up to date.

File: apps/web/package.json [hunk 6]
```diff
@@ -61,6 +72,8 @@
     "recharts": "^2.12.7",
     "satori": "^0.10.14",
     "sharp": "^0.33.4",
+    "three": "^0.168.0",
+    "three-stdlib": "^2.33.0",
```
Comment (JFrankfurt) [inline apps/web/package.json:76]: double checking all these deps ended up getting used and we def want them in the new version

I know sometimes I add a lib and decide against it after working with it for a bit

`maath` looks sus to me when I look it up on npm, since there's no repo, docs, or readme

File: apps/web/public/draco/draco_wasm_wrapper.js (hunk 4 above)
Comment (kirkas) [reply apps/web/public/draco/draco_wasm_wrapper.js:1]: we don't want to load draco JS/WASM from CDN, so we have to set it up locally.

File: apps/web/package.json (hunk 6 above)
Comment (kirkas) [reply apps/web/package.json:76]: yeah we need those. not sure about maath cc @zencephalon ?

File: apps/web/public/draco/draco_wasm_wrapper.js (hunk 4 above)
Comment (JFrankfurt) [reply apps/web/public/draco/draco_wasm_wrapper.js:1]: thanks google

File: apps/web/package.json (hunk 6 above)
Comment (JFrankfurt) [reply apps/web/package.json:75]: all other additions good to merge?

Comment (kirkas) [reply apps/web/package.json:75]: yeah

**Review by kirkas — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for kirkas @ 2024-10-11 19:02:01 UTC
User cannot review their own commit

**Review by omkarb — APPROVED** [review]
